hub/models.py

```diff
@@ -1836,7 +1841,7 @@ async def update_many(self, mapped_records: list[MappedMember], **kwargs):
             "Update many not implemented for this data source type."
         )
 
-    def get_record_id(self, record):
+    def get_record_id(self, record) -> Optional[Union[str, int]]:
```

Comment:
question: lots of things break if get_record_id returns `None`. Can we make this non-optional? if so, what do we return instead of `None`?

Reply:
We could do some heuristic work to generate our own IDs, but then what if the ID in the system changes? Then suddenly we'll have duplicate data in Mapped.

I think the best approach is to keep `get_record_id` honest and then deal with None IDs with the principle that we can't use them, basically. Will submit a separate PR for that.